lib/YaST/Module.pm

```diff
@@ -15,24 +15,32 @@ use strict;
 use warnings;
 use testapi;
 use YuiRestClient;
+use y2_module_basetest qw(wait_for_exit);
 use y2_module_guitest qw(launch_yast2_module_x11);
 use y2_module_consoletest qw(yast2_console_exec);
 
 =head2 open($args)
 
-   open(module => $module, ui => $ui);
+   open {
+     ...
+     ...
+   } {module => $module, ui => $ui};
 
 Unified method to open YaST module while using LibyuiClient. Allows to open the module with both ncurses and Qt UI.
 After starting the module ensures that connection to libyui-rest-api server is established.
+Once the module is opened, run the CODEREF specified between curly braces (or `sub { }` can also be used).
+Finally after running the CODEREF, ensure the module has exited by checking the serial output.
 
 C<$module> - Module to open. e.g. lan, storage, partitioner etc.
             (full list of available yast2 modules can be observed by running 'yast2 -l' in console);
 C<$ui> - User interface the module is expected to be opened with (Possible values: ncurses, qt).
 
 =cut
 
-sub open {
+sub open (&@) {
+    my $code = \&{shift @_};
```

Comment:
Why not to use named argument for the code reference?
I'm also not 100% sure if we should introduce such a wrapper, but if we do, I believe that `open` doesn't reflect what method does. So I would either keep it simple to read and use and have `open` and `close` methods which are used when necessary or rename this method to `execute_actions_in_yast_module` or something in this direction.